src/ngraph/op/fused/clamp.cpp

```diff
@@ -18,12 +18,62 @@
 #include "ngraph/builder/make_constant.hpp"
 #include "ngraph/op/maximum.hpp"
 #include "ngraph/op/minimum.hpp"
+#include "ngraph/runtime/reference/clamp.hpp"
 
 using namespace std;
 using namespace ngraph;
 
 constexpr NodeTypeInfo op::Clamp::type_info;
 
+namespace
+{
+    template <element::Type_t ET>
+    bool evaluate(
+        const HostTensorPtr& arg, const HostTensorPtr& out, double min, double max, size_t count)
+    {
+        runtime::reference::clamp(
+            arg->get_data_ptr<ET>(), out->get_data_ptr<ET>(), min, max, count);
+        return true;
+    }
+
+    bool evaluate_clamp(
+        const HostTensorPtr& arg, const HostTensorPtr& out, double min, double max, size_t count)
+    {
+        bool rc = true;
+        switch (arg->get_element_type())
+        {
+            TYPE_CASE(i8)(arg, out, min, max, count);
+            break;
+            TYPE_CASE(i16)(arg, out, min, max, count);
+            break;
+            TYPE_CASE(i32)(arg, out, min, max, count);
+            break;
+            TYPE_CASE(i64)(arg, out, min, max, count);
+            break;
+            TYPE_CASE(u8)(arg, out, min, max, count);
+            break;
+            TYPE_CASE(u16)(arg, out, min, max, count);
+            break;
+            TYPE_CASE(u32)(arg, out, min, max, count);
+            break;
+            TYPE_CASE(u64)(arg, out, min, max, count);
+            break;
+            TYPE_CASE(f32)(arg, out, min, max, count);
+            break;
+            TYPE_CASE(f64)(arg, out, min, max, count);
+            break;
+        default: rc = false; break;
+        }
+        return rc;
+    }
+}
+
+bool op::v0::Clamp::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs)
+{
+    return evaluate_clamp(
+        inputs[0], outputs[0], get_min(), get_max(), shape_size(get_output_shape(0)));
```

Comment:
output shape may still be partial shape here. I think you need to use the input shape for getting the shape_size here.

Reply:
Input shape == output shape so I don't think I can rely on input shape to solve dynamic shape problems, but I take your point that I need to handle a dynamic shape scenario.

Reply:
When evaluate is called, the input tensor has the static info, the input shape on input tensor is static at this point.

Reply:
Ahh, thanks.  I learned something new today.  I was going to dig into the code and figure all that out.

Reply:
@yimeisun123 I added some dynamic tests cases and left `get_output_shape` here expecting it to fail.  It didn't.  Not sure what this means.

Reply:
UPDATE:  Had some tests fail with commit 7232e30 that do not fail for me locally.  Not sure if that is due to my build config or what.  Experimenting with get_intput_shape() in commit a52d5b6 while I am debugging.

Reply:
Fixed.